plugins/content/joomla/joomla.php

```diff
@@ -615,12 +615,6 @@ public function onContentChangeState($context, $pks, $value)
 			return true;
 		}
 
-		// Check if this function is enabled.
-		if (!$this->params->def('email_new_stage', 0) || $context != 'com_content.article')
```

Comment:
You still need the $context check here.

Reply:
Workflow should not be restricted to com_content.article in the future. So need another criterion.